docs/proposals/karmada-operator/custom-leaf-cert-validity/README.md

```diff
@@ -0,0 +1,77 @@
+---
+title: Support to Configure Leaf Certificate Validity Period in Karmada Operator
+authors:
+- "@jabellard"
+reviewers:
+- "@RainbowMango"
+approvers:
+- "@RainbowMango"
+
+creation-date: 2025-02-28
+
+---
+
+# Support to Configure Leaf Certificate Validity Period in Karmada Operator
+
+## Summary
+
+This proposal aims to extend the Karmada operator with support for configuring the validity period of leaf certificates (e.g., API server certificate). By allowing users to specify the validity period in days, this feature provides flexibility to align with organizational security policies and certificate management practices.
+
+## Motivation
+
+In scenarios where security policies require frequent rotation of certificates, the ability to configure the validity period of leaf certificates is essential. This feature ensures that the Karmada control plane can adhere to these policies by allowing users to specify a custom validity period for leaf certificates.
+
+### Goals
+
+- Allow users to specify the validity period of leaf certificates in days.
+- Ensure that the configuration option is optional and defaults to the current behavior if not specified.
+- Enable operators to align Karmada control plane PKI with organizational security policies.
+
+### Non-Goals
+
+- Change the default behavior of the Karmada operator when no custom validity period is provided.
+
+## Proposal
+
+The proposal introduces a new optional field, `LeafCertValidity`, in the `CustomCertificate` struct, where users can specify the validity period of leaf certificates in days.
+
+### API Changes
+
+```go
+type CustomCertificate struct {
+   // Other, existing fields omitted for brevity
+
+   // LeafCertValidity specifies the validity period of leaf certificates (e.g., API Server certificate) in days.
+   // If not specified, the default validity period of 1 year will be used.
+   // +optional
+   LeafCertValidity *int `json:"leafCertValidity,omitempty"`
```

Comment:
This configuration would apply to all certificates that are issued by the `APIServerCACert` above, right?

Reply:
By the way, is the `LeafCert` a well-known or standard terminology? What's the unit of this filed, days?

Reply:
When provisioning a Karmada instance, we make use of these CAs:
- API server CA
- Front Proxy CA
- ETCD CA (when using in-cluster ETCD)

Currently the validity period of all the certs signed by those CAs is the same and default to 1 year. As such, this configuration is intended to apply to all those certs, not just the ones signed by the API server CA.

As far as I'm aware, the terms leaf certificate and end-entity certificate are what's usually used to describe the certificate's role in a chain of trust. I chose to go with leaf certificate as [that aligns very well with the terminology used in other well established projects like cert-manager](https://cert-manager.io/docs/reference/tls-terminology/#leaf-certificates).

Reply:
The unit is days. I added that in the comments.

Reply:
Thanks for the update. I got it.

How about renaming the field to `LeafCertValidityDays` to clearly indicate that the value is specified in days?

Reply:
Just pushed changes to update.

Reply:
@RainbowMango , any other comments? By the way, I just pushed a [PR](https://github.com/karmada-io/karmada/pull/6193) for the implementation.